easybuild/easyblocks/a/aomp.py

```diff
@@ -154,3 +156,38 @@ def post_install_step(self):
                       " or isn't a directory!"
             raise EasyBuildError(err_str.format(actual_install,
                                                 self.installdir))
+
+    def sanity_check_step(self):
+        """Custom sanity check for AOMP"""
+        shlib_ext = get_shared_lib_ext()
+        arch = get_cpu_architecture()
+        # Check architecture explicitly since Clang uses potentially
+        # different names
+        if arch == X86_64:
+            arch = 'x86_64'
+        elif arch == POWER:
+            arch = 'ppc64'
+        elif arch == AARCH64:
+            arch = 'aarch64'
```

Comment:
this can be simplified a bit:
```suggestion
        arch_map = {
            X86_64: 'x86_64',
            POWER: 'ppc64',
            AARCH64: 'aarch64',
        }

        if arch in arch_map:
            arch = arch_map[arch]
```